CMakeLists.txt

```diff
@@ -333,7 +333,37 @@ if(GCC OR CLANG)
   # TODO(CryptoAlg-759): enable '-Wpedantic' if awslc has to follow c99 spec.
   if(CLANG OR (GCC AND CMAKE_C_COMPILER_VERSION VERSION_GREATER "4.1.3"))
     # GCC 4.1.3 and below do not support all of these flags or they raise false positives.
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fvisibility=hidden -Wall -Wextra -Wno-unused-parameter -Werror")
+    if(MSVC)
+      set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-bad-function-cast")
+      set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-cast-function-type-strict")
+      set(C_CXX_FLAGS "${C_CXX_FLAGS} -Wno-cast-function-type")
+      set(C_CXX_FLAGS "${C_CXX_FLAGS} -Wno-cast-align")
+      set(C_CXX_FLAGS "${C_CXX_FLAGS} -Wno-cast-qual")
+      set(C_CXX_FLAGS "${C_CXX_FLAGS} -Wno-comma")
+      set(C_CXX_FLAGS "${C_CXX_FLAGS} -Wno-c++98-compat")
+      set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-declaration-after-statement")
+      set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-disabled-macro-expansion")
+      set(C_CXX_FLAGS "${C_CXX_FLAGS} -Wno-deprecated-declarations")
+      set(C_CXX_FLAGS "${C_CXX_FLAGS} -Wno-extra-semi-stmt")
+      set(C_CXX_FLAGS "${C_CXX_FLAGS} -Wno-extra-semi")
+      set(C_CXX_FLAGS "${C_CXX_FLAGS} -Wno-implicit-int-conversion")
+      set(C_CXX_FLAGS "${C_CXX_FLAGS} -Wno-missing-noreturn")
+      set(C_CXX_FLAGS "${C_CXX_FLAGS} -Wno-missing-variable-declarations")
+      set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-overlength-strings")
+      set(C_CXX_FLAGS "${C_CXX_FLAGS} -Wno-nonportable-system-include-path")
+      set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-reserved-identifier")
+      set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-shorten-64-to-32")
+      set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-sign-conversion")
+      set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-unreachable-code-break")
+      set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-tautological-value-range-compare")
+      set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-unreachable-code-return")
+      set(C_CXX_FLAGS "${C_CXX_FLAGS} -Wno-unsafe-buffer-usage")
+      set(C_CXX_FLAGS "${C_CXX_FLAGS} -Wno-unused-macros")
+      set(C_CXX_FLAGS "${C_CXX_FLAGS} -Wno-used-but-marked-unused")
```

Comment:
Is there a reason why we're appending to the flags like this? These seem to be all over the place and it'd be cleaner if we could define these like our other flags.
Ex:
```
set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-bad-function-cast -Wno-cast-function-type-strict -Wno-declaration-after-statement -Wno-disabled-macro-expansion ...")
```

Reply:
This format makes it easier to disable specific flags (i.e., by commenting out a line) when working to correct these warnings. For example, I've got a local branch where I've started addressing `-Wno-nonportable-system-include-path`. There are a few other Warnings that appear to be quite fixable with just a little effort.  I plan to periodically post PRs that address such warnings.

Reply:
Sure, if this does makes things easier I guess there's no harm keeping it. Should we leave a TODO comment above this that we'll get to these turned off warnings and group them together? And for now, could we at least group the `CMAKE_C_FLAGS` and `C_CXX_FLAGS` in the same sections?